### PR TITLE
Add AFS Groovy idiomatic builder

### PR DESCRIPTION
**Pull request by geofjamg**

### CODE REVIEW

File: scripting/src/main/groovy/com/powsybl/scripting/groovy/GroovyScripts.groovy [hunk 1]
```diff
@@ -6,17 +6,31 @@
...
 
 import java.nio.charset.StandardCharsets
 import java.nio.file.Path
 
+class BuilderSpec {
+
+    private final def builder
+
+    BuilderSpec(Object builder) {
+        this.builder = builder
```
Comment (mathbagu) [inline scripting/src/main/groovy/com/powsybl/scripting/groovy/GroovyScripts.groovy]: Assert that builder is not null

File: scripting/src/main/groovy/com/powsybl/scripting/groovy/GroovyScripts.groovy [hunk 2]
```diff
@@ -6,17 +6,31 @@
...
+    }
+
+    def methodMissing(String name, args) {
```
Comment (mathbagu) [inline scripting/src/main/groovy/com/powsybl/scripting/groovy/GroovyScripts.groovy]: Assert that name is not null

File: scripting/src/test/java/com/powsybl/scripting/groovy/AfsExtensionGroovyScriptTest.java [hunk 3]
```diff
@@ -35,15 +35,27 @@ protected AppFileSystemStorage createStorage() {
...
+                "    name 'foo2'",
+                "}",
                 "print foo2.getName()"
         ));
     }
 
     @Override
     protected String getExpectedOutput() {
-        return "foo" + System.lineSeparator() + "foo";
+        return "foo" + System.lineSeparator() +
```
Comment (mathbagu) [inline scripting/src/test/java/com/powsybl/scripting/groovy/AfsExtensionGroovyScriptTest.java]: String.join(System.lineSeparator(), "foo", "foo", "foo2")

File: scripting/src/main/groovy/com/powsybl/scripting/groovy/GroovyScripts.groovy (hunk 2 above)
Comment (geofjamg) [reply scripting/src/main/groovy/com/powsybl/scripting/groovy/GroovyScripts.groovy]: This method is called by Groovy framework when trying to call a missing method, so there is no no need to check anything

File: scripting/src/main/groovy/com/powsybl/scripting/groovy/GroovyScripts.groovy (hunk 1 above)
Comment (geofjamg) [reply scripting/src/main/groovy/com/powsybl/scripting/groovy/GroovyScripts.groovy]: Done

File: scripting/src/test/java/com/powsybl/scripting/groovy/AfsExtensionGroovyScriptTest.java (hunk 3 above)
Comment (geofjamg) [reply scripting/src/test/java/com/powsybl/scripting/groovy/AfsExtensionGroovyScriptTest.java]: Done

**Pull request merged**